recipes/cppcmd/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from glob import glob
+
+
+class CppCmdConan(ConanFile):
+    name = "cppcmd"
+    description = "Simple cpp command interpreter header-only library"
+    topics = ("header-only", "interpreter", "cpp")
+    homepage = "https://github.com/remysalim/cppcmd"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _supports_cpp17(self):
+        supported_compilers = [
+            ("gcc", "8"), ("clang", "7"), ("apple-clang", "10.2"), ("Visual Studio", "15.7")]
+        compiler = self.settings.compiler
+        version = tools.Version(compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        elif not self._supports_cpp17():
+            raise ConanInvalidConfiguration("cppcmd requires C++17 support")
```

Comment:
We typically don't like to restrict "unknown" compilers which I believe would fail (maybe, your python is much better than mine)

Could you please allow this scenario and print a warning? It would be greatly appreciated 😸 

I did this a while back, https://github.com/conan-io/conan-center-index/blob/master/recipes/skyr-url/all/conanfile.py#L50-L58 and there's more of this copy pasted but it's a good example

Reply:
You're absolutely right, it would fail on an unknown compiler, I'm copy pasting your snippet instead, thanks!